nss_cache/sources/ldapsource.py

```diff
@@ -397,9 +397,15 @@ def __iter__(self):
         # If the dn is requested, return it along with the payload,
         # otherwise ignore it.
         for key in record[1]:
-          if isinstance(record[1][key][0], bytes) and key != 'objectSid':
-            value = record[1][key][0].decode('utf-8')
-            record[1][key] = [value]
+          if len(record[1][key]) > 1:
+            for i, item in enumerate(record[1][key]):
```

Comment:
Then i think we can avoid the enumerate() line by juster iterating over record[1][key],

Reply:
this will fail the `runtests.py` as far as I remember.

Reply:
if ` record[1][key]` is a string this will not work, right?

Reply:
we need to consider, that there one element lists, where the iteration will be over the string characters instead of the list elements. So applying the following:
```
for item in record[1][key]:
```
Won't work on [b'Test User']:
will iterate over

Reply:
may be using `range` is a better way?